### ISSUE TITLE
[F-Req] Automatic window shift when using keyboard

### ISSUE THREAD

**Issue by Felix14-v2** [PojavLauncherTeam/PojavLauncher#4627]
### Description

Using a keyboard in PojavLauncher at the current state is a bit painful because of inability to see what you write if the keyboard overlaps an input field (chat or modded search fields, for example).

### Alternatives

HMCL-PE/FoldCraft Launchers shifts the game window according the keyboard height, but this also isn't an option in my opinion, because you can't see the text in some fields with this approach, such as search bar in the world screen or in creative inventory.
![Screenshot_20230924-050333](https://github.com/PojavLauncherTeam/PojavLauncher/assets/75726196/3eb1bb62-dc21-41be-87f0-773a7060e71f)

### Proposed solution

In most cases we need to focus the text field before writing in it. So the best solution (imo) is to store coordinates of the last point on the screen clicked by user and shift the window taking into account ordinate of that point when the keyboard opened.

Another weird solution would be a separate button for custom controls to open keyboard specifically with the window shifting. But yeah, this is a bit weird.

**Comment by Felix14-v2** [comment]
Oh, I used the wrong template... someone, please fix the labels :D

**Comment by MovTery** [comment]
You can try an alternative keyboard, such as one that supports levitation and can freely change the opacity. You can also buy a physical keyboard and mouse and connect it to your device via Bluetooth or OTG.![Screenshot_2023-09-24-11-26-31-31_d6aef70892487175e4504e12eca5bc09.jpg](https://github.com/PojavLauncherTeam/PojavLauncher/assets/123115192/01271117-f4db-4956-b6db-a04b7756b967)

**Comment by Felix14-v2** [comment]
> You can try an alternative keyboard, such as one that supports levitation and can freely change the opacity

Yeah, this is the only workaround we can use for now. But... still not the best solution, because GBoard and other keyboards with this feature don't work well with my language.

> You can also buy a physical keyboard and mouse and connect it to your device via Bluetooth or OTG

Unfortunately not all devices supports OTG. Moreover, there are devices that don't support the Bluetooth keyboard option (mine fore example). 

Also it is inconvenient to use a physical keyboard when you go to school or work by bus or underground :)

**Comment by GaucovyAero** [comment]
the hmcl way of doing it could be added i think

**Comment by Felix14-v2** [comment]
HMCL implementation doesn't allow you to see the vast majority of fields, such as inventory and recipe book search, world screen search, world name and seed, server name and IP, some fields at the gamerules screen, and others, including modded ones. I don't think this is a good option.

**Comment by GaucovyAero** [comment]
then i think you gotta get a transparent or floating keyboard

**Comment by GaucovyAero** [comment]
@HopiHopy what virtual keyboard are you using btw?

**Comment by MovTery** [comment]
> @HopiHopy what virtual keyboard are you using btw?

This is a popular keyboard in China, but I'm not sure if it's available for download in other countries![IMG_20230925_044120.jpg](https://github.com/PojavLauncherTeam/PojavLauncher/assets/123115192/47138a60-037d-4d28-8035-3876ea6c11f3)

**Comment by ilikehotpotateos** [comment]
> > @HopiHopy what virtual keyboard are you using btw?
> 
> This is a popular keyboard in China, but I'm not sure if it's available for download in other countries![IMG_20230925_044120.jpg](https://user-images.githubusercontent.com/123115192/270190879-47138a60-037d-4d28-8035-3876ea6c11f3.jpg)

Will I get all my data stolen?

**Comment by GaucovyAero** [comment]
alr thanks

**Comment by shestako2** [comment]
Отличная идея
